### DISCUSSION TITLE
Node Public Interfaces Team Weekly Update - 2021-07-01

### DISCUSSION THREAD

**Discussion by khorolets** [near/nearcore#4454] [Node Public Interfaces]
*While @frol is on vacation @khorolets writes this report so it might be not so great*

Last week update is here: #4417

# Brief Summary

Last week the team was focused on different areas: prepare and support future release of 1.20.0, reliability of the data, and availability of the services. 

# Results

* [nearcore](https://github.com/near/nearcore)
  * Exposed structured errors from JSON RPC merged and will be included in 1.20.0 (@khorolets)
     #4399
  * Introduced strictly typed `AccountId` type throughout workspace to enforce guaranteed validation is under review (@miraclx)
     #4440
  * Implemented `--download-config` option in the `init` command (@chefsale)
     #4427
  * Added a stale bot to make sure issues and PRs get closed or responded to (@chefsale)
     https://github.com/near/nearcore/commit/5a81fa01ba9402b02251e835699c75f9bb98b516
  * fixed bulldozer automerge due to label changes in nearcore (@chefsale)

* [indexer-for-explorer](https://github.com/near/near-indexer-for-explorer)
  * `account_changes` table finally has no duplicates (@khorolets @telezhnaya)
     https://github.com/near/near-indexer-for-explorer/issues/74
  * Draft PR which fixes null-bytes issue https://github.com/near/near-indexer-for-explorer/pull/131 is created. During the attempt to test it on historical data an error in nearcore was noticed https://github.com/near/nearcore/issues/4441 (@khorolets)

* [near-cli-rs](https://github.com/FroVolod/near-cli)
  * Created a stub for the project homepage: https://near.cli.rs (currently it redirects to the near-cli-rs repository) (@frol)

* [near-sdk-rs](https://github.com/near/near-sdk-rs) 
  * Found proper configuration for building lockup contract (@telezhnaya)
     https://github.com/near/near-sdk-rs/pull/454

* [explorer](https://github.com/near/near-explorer)
  * Display active access keys on Account Details page(@shelegdmitriy)
     https://github.com/near/near-explorer/pull/675
  * Finalized final stage of comparing Validators and Proposals into single tab (@shelegdmitriy)
     https://github.com/near/near-explorer/pull/661

* operations
  * testnet deploy with some debugging (@mhalambek)
  * oncall stuff, betanet crash as the major one (@mhalambek)

* meta
  * worked on getting betanet working today and yesterday due to one validator not being synced properly after hardfork (@chefsale)

# Challenges & Blockers:

* indexer-for-explorer
  * Data consistency due to the old bug leading to some transaction being applied on mainnet twice is still not resolved on Indexer for Explorer side
    https://github.com/near/near-indexer-for-explorer/issues/84#issuecomment-857615497

# Action Items:

* nearcore
  * Publish `near-account-id` as an independent crate to be used throughout the ecosystem. (@miraclx)
  * Work on getting reusable core crates (e.g `near-primitives`, `near-crypto`) lightweight and publishable (@miraclx)
  * Working on implementing `--boot-nodes` option in the init command, waiting to land the `--download-config` first (@chefsale)

* indexer-for-explorer
  * Calculate historical circulating supply and store it in indexer DB (@telezhnaya)
    https://github.com/near/near-indexer-for-explorer/commits/olya/circulating
  * Create a new release for Indexer for Explorer compatible with 1.20.0 (@khorolets)
  * Finish null-bytes issue (@khorolets)
     https://github.com/near/near-indexer-for-explorer/pull/131

* lockup:
  *Fix lockup contract comments from @ailisp report https://hackmd.io/@ailisp/rJbiDYr5_#foundation_callbacksrs https://github.com/near/core-contracts/compare/olya/fix-lockup-after-review (@telezhnaya)

* docs
  * Document the data flowing in NEAR Protocol (@khorolets)
    https://github.com/near/docs/issues/700

* explorer
  * Expose average fees in the last 24 hours and 7 days through API (@shelegdmitriy)
     https://github.com/near/near-explorer/issues/627

* operations
  * Rolling out new alert system to betanet and mainnet (@mhalambek)
  * Maybe adding some pagerduty config along with it (@mhalambek)

**Comment by maxzaver** [comment]
@DiscRiskandBisque  @frol what is the decision regarding near-cli-rs and near-cli? Are we merging the two? Should we wait with any changes to near-cli-rs before the decision is made?

**Reply by frol** [reply]
We will sync up with Josh on 2021-07-26 (after he is back from vacation)
